sample-app/src/hooks/useNearMeSearch.ts

```diff
@@ -0,0 +1,41 @@
+import { AnswersHeadless } from '@yext/answers-headless-react';
+import { executeSearch, updateLocationIfNeeded } from '../utils/search-operations';
+import { MutableRefObject, useRef } from 'react';
+import { AutocompleteResponse, SearchIntent } from '@yext/answers-headless-react';
+
+type QueryFunc = () => Promise<void>
+type AutocompleteRef = MutableRefObject<Promise<AutocompleteResponse | undefined> | undefined>
+
+/**
+ * Returns a search action that will handle near me searches, by first checking
+ * for near me intents using an autocomplete request.
+ * You can optionally use the provided ref to store autocomplete responses, to avoid
+ * making unnecessary autocomplete requests.
+ */
+export default function useNearMeSearch(
```

Comment:
This seems like something @yen-tt wanted to add in a previous PR. We ultimately decided against having a single method with all this responsibility. See the relevant comments here:
- https://github.com/yext/answers-headless-react/pull/57#discussion_r752288311
- https://github.com/yext/answers-headless-react/pull/57#discussion_r752293077